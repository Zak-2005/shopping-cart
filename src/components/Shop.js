import Card from "./Card";
import Filter from "./Filter";
import { products } from "./ProductList";
import {  useState } from "react";
const Shop = () => {
  const [utensilFilter, setUtensilFilter] = useState(false);
  const [dishwareFilter, setDishwareFilter] = useState(false);
  const [napkinFilter, setNapkinFilter] = useState(false);
  return (
    <div className="shopContent">
      <div className="filter">
      <h1 id="shopHeading" test-id="shop">Shop</h1>
      <Filter
          label="Utensils"
          setChecked={setUtensilFilter}
        />
        <Filter
          label="Dishware"
          setChecked={setDishwareFilter}
        />
        <Filter
          label="Napkins"
          setChecked={setNapkinFilter}
        />
      </div>
      <div className="products">
        {products.map((product, index) => {
          if (!utensilFilter && !dishwareFilter && !napkinFilter) {
            return (
              <Card
                src={`./Images/${product.title}.png`}
                title={product.title}
                price={product.price}
                key={index}
              />
            );
          } else if (utensilFilter && product.type === "Utensil") {
            return (
              <Card
                src={`./Images/${product.title}.png`}
                title={product.title}
                price={product.price}
                key={index}
              />
            );
          } else if (dishwareFilter && product.type === "Dishware") {
            return (
              <Card
                src={`./Images/${product.title}.png`}
                title={product.title}
                price={product.price}
                key={index}
              />
            );
          } else if (napkinFilter && product.type === "Napkin") {
            return (
              <Card
                src={`./Images/${product.title}.png`}
                title={product.title}
                price={product.price}
                key={index}
              />
            );
          }
        })}
      </div>
    </div>
  );
};

export default Shop;
