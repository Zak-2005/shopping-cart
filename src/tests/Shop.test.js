import React from "react";
import { render, screen } from "@testing-library/react";
import Shop from "../components/Shop";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import { act } from "react-dom/test-utils";
import { HashRouter } from "react-router-dom";
describe("Shop Component", ()=>{
    it("Shop Renders All products and filters on load", ()=>{
        const {container}= render(<HashRouter><Shop /></HashRouter>)
        expect(container).toMatchSnapshot()
    })
    it("Utensil Filter only renders Utensils", async ()=>{
        const user = userEvent.setup();
        render(<HashRouter><Shop /></HashRouter>)

        const utensilFilter = screen.getByPlaceholderText("Utensils")
       await act(async ()=> { await user.click(utensilFilter)})
        

        expect(screen.queryByText("Bowl")).not.toBeInTheDocument()
    })
    it("Dishware Filter only renders Dishware", async ()=>{
        const user = userEvent.setup();
        render(<HashRouter><Shop /></HashRouter>)

        const dishwareFilter = screen.getByPlaceholderText("Dishware")
       await act(async()=> { await user.click(dishwareFilter)})
        

        expect(screen.queryByText("Spork")).not.toBeInTheDocument()
    })
    it("Napkins Filter only renders Napkins", async ()=>{
        const user = userEvent.setup();
        render(<HashRouter><Shop /></HashRouter>)

        const napkinFilter = screen.getByPlaceholderText("Napkins")
       await act(async()=> { await user.click(napkinFilter)})
        

        expect(screen.queryByText("Spork")).not.toBeInTheDocument()
    })
})