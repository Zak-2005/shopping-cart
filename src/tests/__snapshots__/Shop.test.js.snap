// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop Component Shop Renders All products and filters on load 1`] = `
<div>
  <div
    class="shopContent"
  >
    <div
      class="filter"
    >
      <h1
        id="shopHeading"
        test-id="shop"
      >
        Shop
      </h1>
      <div
        class="filterText"
      >
        <input
          placeholder="Utensils"
          type="checkbox"
        />
        <label>
          Utensils
        </label>
      </div>
      <div
        class="filterText"
      >
        <input
          placeholder="Dishware"
          type="checkbox"
        />
        <label>
          Dishware
        </label>
      </div>
      <div
        class="filterText"
      >
        <input
          placeholder="Napkins"
          type="checkbox"
        />
        <label>
          Napkins
        </label>
      </div>
    </div>
    <div
      class="products"
    >
      <div
        class="card"
      >
        <img
          alt="./Images/Exquisite-Spork.png"
          class="cardImg"
          src="./Images/Exquisite-Spork.png"
        />
        <h4>
          <a
            href="#/shop/Exquisite-Spork"
          >
            Exquisite-Spork
          </a>
        </h4>
        <h4>
          $125
        </h4>
      </div>
      <div
        class="card"
      >
        <img
          alt="./Images/Exquisite-Splayd.png"
          class="cardImg"
          src="./Images/Exquisite-Splayd.png"
        />
        <h4>
          <a
            href="#/shop/Exquisite-Splayd"
          >
            Exquisite-Splayd
          </a>
        </h4>
        <h4>
          $155
        </h4>
      </div>
      <div
        class="card"
      >
        <img
          alt="./Images/Wooden-Spoon.png"
          class="cardImg"
          src="./Images/Wooden-Spoon.png"
        />
        <h4>
          <a
            href="#/shop/Wooden-Spoon"
          >
            Wooden-Spoon
          </a>
        </h4>
        <h4>
          $15
        </h4>
      </div>
      <div
        class="card"
      >
        <img
          alt="./Images/Blate.png"
          class="cardImg"
          src="./Images/Blate.png"
        />
        <h4>
          <a
            href="#/shop/Blate"
          >
            Blate
          </a>
        </h4>
        <h4>
          $165
        </h4>
      </div>
      <div
        class="card"
      >
        <img
          alt="./Images/Bowl.png"
          class="cardImg"
          src="./Images/Bowl.png"
        />
        <h4>
          <a
            href="#/shop/Bowl"
          >
            Bowl
          </a>
        </h4>
        <h4>
          $54
        </h4>
      </div>
      <div
        class="card"
      >
        <img
          alt="./Images/Plate.png"
          class="cardImg"
          src="./Images/Plate.png"
        />
        <h4>
          <a
            href="#/shop/Plate"
          >
            Plate
          </a>
        </h4>
        <h4>
          $54
        </h4>
      </div>
      <div
        class="card"
      >
        <img
          alt="./Images/White-Napkins.png"
          class="cardImg"
          src="./Images/White-Napkins.png"
        />
        <h4>
          <a
            href="#/shop/White-Napkins"
          >
            White-Napkins
          </a>
        </h4>
        <h4>
          $20
        </h4>
      </div>
      <div
        class="card"
      >
        <img
          alt="./Images/Blue-Napkins.png"
          class="cardImg"
          src="./Images/Blue-Napkins.png"
        />
        <h4>
          <a
            href="#/shop/Blue-Napkins"
          >
            Blue-Napkins
          </a>
        </h4>
        <h4>
          $20
        </h4>
      </div>
      <div
        class="card"
      >
        <img
          alt="./Images/Green-Napkins.png"
          class="cardImg"
          src="./Images/Green-Napkins.png"
        />
        <h4>
          <a
            href="#/shop/Green-Napkins"
          >
            Green-Napkins
          </a>
        </h4>
        <h4>
          $20
        </h4>
      </div>
    </div>
  </div>
</div>
`;
